### PR TITLE
Add a web page implementing a simple WebIDL checker.

### PR DESCRIPTION
**Pull request by kenrussell**

### CODE REVIEW

**Comment by kenrussell** [comment]
Note: I'll be happy to point the README to github.io/w3c/webidl2.js/checker/ if you wouldn't mind setting up Github pages for this repository. According to https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/ it looks like there is now the option of serving up Github pages from the master branch, so if you'd perhaps consider moving the "develop" branch to "master" then pull requests into this repo would go live on the web site. Thanks for considering this.

**Comment by kenrussell** [comment]
Addressed review feedback; added another textarea with the JSON.stringify'd result of the parser, optionally pretty-printed. Please take another look.

**Comment by kenrussell** [comment]
Could someone take another look at this? Thanks.

**Comment by marcoscaceres** [comment]
@saschanaz, I'm guess we could use this as a base to build onto of? It's rather basic - and markup could definitely be beefed up a bit (e.g., should really use a proper `<form>`, and `<output>` element, and a nice stylesheet, etc.)... I don't have a strong opinion tho.

File: checker/index.html [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  if (parserResult) {
+    let prettyPrintEl = document.getElementById('pretty-print');
+    outputEl.innerText = JSON.stringify(parserResult, null, prettyPrintEl.checked ? 2 : null);
+  } else {
+    outputEl.innerText = '';
+  }
+}
+
+function checkWebIDL() {
+  let validation = document.getElementById('webidl-checker-validation');
```
Comment (saschanaz) [inline checker/index.html]: nit: const instead of let

File: checker/index.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
+<!DOCTYPE html><html><head><title>WebIDL 2 Checker</title>
+<script src='../lib/webidl2.js'></script>
+<script type='text/javascript'>
+let parserResult = undefined;
+
+function formatParserOutput() {
+  let outputEl = document.getElementById('webidl-checker-output');
```
Comment (saschanaz) [inline checker/index.html]: nit: const instead of let

File: checker/index.html [hunk 3]
```diff
@@ -0,0 +1,65 @@
+<!DOCTYPE html><html><head><title>WebIDL 2 Checker</title>
+<script src='../lib/webidl2.js'></script>
+<script type='text/javascript'>
```
Comment (saschanaz) [inline checker/index.html]: nit: text/javascript is not required since HTML5

**Comment by saschanaz** [comment]
I think this is okay to start.

File: checker/index.html [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+let parserResult = undefined;
+
+function formatParserOutput() {
+  let outputEl = document.getElementById('webidl-checker-output');
+  if (parserResult) {
+    let prettyPrintEl = document.getElementById('pretty-print');
```
Comment (marcoscaceres) [inline checker/index.html]: const here too.

File: checker/index.html [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+</p>
+<textarea id='webidl-checker-validation' rows='20' cols='80'>
+</textarea>
+<p>
+Parser output:
+</p>
+<textarea id='webidl-checker-output' rows='20' cols='80'>
+</textarea>
+<br>
+<input type='checkbox' id='pretty-print' checked='true' onchange='formatParserOutput()'>Pretty Print</button>
```
Comment (marcoscaceres) [inline checker/index.html]: stray `</button>`?

File: checker/index.html [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+<button onclick='checkWebIDL()'>Check WebIDL</button>
+<p>
+Validation results:
+</p>
+<textarea id='webidl-checker-validation' rows='20' cols='80'>
+</textarea>
+<p>
+Parser output:
+</p>
+<textarea id='webidl-checker-output' rows='20' cols='80'>
```
Comment (marcoscaceres) [inline checker/index.html]: Why not just use a pre here?

**Comment by marcoscaceres** [comment]
Might also be good to run the markup over HTML tidy or http://jsbeautifier.org

File: checker/index.html (hunk 1 above)
Comment (kenrussell) [reply checker/index.html]: Done.

File: checker/index.html (hunk 2 above)
Comment (kenrussell) [reply checker/index.html]: Done.

File: checker/index.html (hunk 3 above)
Comment (kenrussell) [reply checker/index.html]: Removed.

File: checker/index.html (hunk 4 above)
Comment (kenrussell) [reply checker/index.html]: Done.

File: checker/index.html (hunk 5 above)
Comment (kenrussell) [reply checker/index.html]: Thanks, fixed.

File: checker/index.html (hunk 6 above)
Comment (kenrussell) [reply checker/index.html]: I wanted the behavior of the textarea – displaying this number of rows and columns, but allowing scrolling – and couldn't figure out how to style a pre element to do that.

**Comment by kenrussell** [comment]
Ran through the current version of https://github.com/htacg/tidy-html5 .

File: checker/index.html [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  const outputEl = document.getElementById('webidl-checker-output');
+  if (parserResult) {
+    const prettyPrintEl = document.getElementById('pretty-print');
+    outputEl.innerText = JSON.stringify(parserResult, null, prettyPrintEl.checked ? 2 : null);
+  } else {
+    outputEl.innerText = '';
+  }
+}
+
+function checkWebIDL() {
```
Comment (marcoscaceres) [inline checker/index.html]: checkWebIDL() should only take a `textToCheck`.

File: checker/index.html [hunk 8]
```diff
@@ -0,0 +1,56 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta name="generator" content="HTML Tidy for HTML5 for Apple macOS version 5.7.16">
```
Comment (marcoscaceres) [inline checker/index.html]: nit: please remove this

File: checker/index.html [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+<title>WebIDL 2 Checker</title>
+
+<script src='../lib/webidl2.js'></script>
+<script>
+let parserResult = undefined;
```
[the remark this answers is not in the document]
Comment (kenrussell) [reply checker/index.html]: It's necessary in order to let the "Pretty Print" checkbox work without re-parsing the IDL. I wouldn't want to make that checkbox re-parse it.

File: checker/index.html (hunk 7 above)
Comment (kenrussell) [reply checker/index.html]: Done.

File: checker/index.html (hunk 8 above)
Comment (kenrussell) [reply checker/index.html]: Seems a bit rude to the authors of htmltidy to not give them credit...but done.

**Comment by kenrussell** [comment]
All of the review feedback that could be addressed has been addressed.

**Comment by marcoscaceres** [comment]
Thanks @kenrussell for contributing this. I think it will be useful.

**Pull request merged**